ts/Core/Chart/Chart.ts

```diff
@@ -514,15 +518,17 @@ class Chart {
             this.time = new Time(extend(
                 options.time || {},
                 {
-                    locale: (
-                        options.lang.locale ??
-                        (this.renderTo.closest('[lang]') as HTMLDOMElement|null)
-                            ?.lang
-                    )
+                    locale: this.locale
                 }
             ));
             options.time = this.time.options;
 
+            // Detect the number format of the locale and set the lang settings
+            // eslint-disable-next-line new-cap
+            const numString = Intl.NumberFormat(this.locale).format(10000.1);
+            options.lang.thousandsSep ??= numString[2];
+            options.lang.decimalPoint ??= numString[6];
```

Comment:
I hope there no extraordinary locale that would make 2 and 6 point to something else the what's needed.

Reply:
I tested that with all locales known to my browser, they all came out fine.

Reply:
You're right! I did test this in Chrome, but I had another look now to be sure, and it actually fails in Firefox and Safari. There are some languages that omit the thousands separator, making both `thousandsSep` and `decimalPoint` fail. https://jsfiddle.net/highcharts/f649mv5e/